src/MultilineStrings.jl

```diff
@@ -0,0 +1,253 @@
+
+# Keywords
+- `style::Symbol`: Replace newlines with spaces (`:folded`) or keep newlines (`:literal`)
+- `chomp::Symbol`: No newlines at the end (`:strip`), single newline at the end (`:clip`),
+  or keep all newlines from the end (`:keep`)
+"""
+function multiline(str::AbstractString; style=DEFAULT_STYLE, chomp=DEFAULT_CHOMP)
+    multiline(str, style, chomp)
+end
+
+"""
+    multiline(str, indicators) -> AbstractString
+
+Revise a multiline string according to the provided style and chomp encoded in the
+`indicators` string.
+
+# Arguments
+- `str::AbstractString`: The multiline string to be processed
+- `indicators::AbstractString`: A terse string representing the style and chomp. Indicators
+  can be either in letter-form or in YAML-form:
+
+    - "fs" / ">-": folded and strip
+    - "fc" / ">": folded and clip
+    - "fk" / ">+": folded and keep
+    - "ls" / "|-": literal and strip
+    - "lc" / "|": literal and clip
+    - "lk" / "|+": literal and keep
+"""
+function multiline(str::AbstractString, indicators::AbstractString)
+    indicators_len = length(indicators)
+    indicators_len > 2 && throw(ArgumentError("Too many indicators provided"))
+
+    # Note: Using '\0` to indicate undefined
+    yaml_chomp = false
+    style_char, chomp_char = if indicators_len == 2
+        indicators
+    elseif indicators_len == 1
+        ind = indicators[1]
+        if ind in ('f', 'l')
+            ind, '\0'
+        elseif ind in ('>', '|')
+            yaml_chomp = true
+            ind, '\0'
+        else
+            '\0', ind
+        end
+    else
+        '\0', '\0'
+    end
+
+    if style_char != '\0' && chomp_char != '\0' && isletter(style_char) ⊻ isletter(chomp_char)
+        throw(ArgumentError("Can't mix YAML style block indicators with letter indicators"))
+    end
+
+    style = if style_char == 'f' || style_char == '>'
+        :folded
+    elseif style_char == 'l' || style_char == '|'
+        :literal
+    elseif style_char == '\0'
+        DEFAULT_STYLE
+    else
+        throw(ArgumentError("Unknown style indicator: $(repr(style_char))"))
+    end
+
+    chomp = if chomp_char == 'c' || yaml_chomp
+        :clip
+    elseif chomp_char == 's' || chomp_char == '-'
+        :strip
+    elseif chomp_char == 'k' || chomp_char == '+'
+        :keep
+    elseif chomp_char == '\0'
+        DEFAULT_CHOMP
+    else
+        throw(ArgumentError("Unknown chomping indicator: $(repr(chomp_char))"))
+    end
+
+    return multiline(str, style, chomp)
+end
+
+function multiline(str::AbstractString, style::Symbol, chomp::Symbol)
+    # Append an additional, non-space, character to force one more iteration of the style
+    # loop
+    str *= ETX
+
+    out = IOBuffer()
+    num_newlines = 0  # The number of newlines at the end of the string
+    prev = curr = '\0'
+
+    # Replace newlines with spaces (folded)
+    if style === :folded
+        # The code below is equivalent to these two regexes:
```

Comment:
Why not just use the regex code?

Reply:
The regex is far less performant. The regular expression version
```julia
julia> @btime multiline("a\nb\n\nc\n", "fk")
  751.017 ns (12 allocations: 576 bytes)
"a b\nc\n"
```
Current implementation:
```julia
julia> @btime multiline("a\nb\n\nc\n", "fk")
  223.140 ns (6 allocations: 320 bytes)
"a b\nc\n"
```

Reply:
I can update the comment to say as much